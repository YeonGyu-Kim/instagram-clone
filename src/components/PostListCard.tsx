'use client';
import Image from 'next/image';
import Avatar from './Avatar';
import CommentForm from './CommentForm';
import ActionBar from './ActionBar';
import { SimplePost } from '../../sanity-studio/model/posts';
import { useState } from 'react';
import ModalPortal from './ui/ModalPortal';
import PostModal from './PostModal';
import PostDetail from './PostDetail';

type Props = {
  post: SimplePost;
  priority?: boolean;
};

export default function PostListCard({ post, priority = false }: Props) {
  const { username, userImage, image, likes, text, createdAt } = post;
  const [openModal, setOpenModal] = useState(false);
  return (
    <article className='rounded-lg border border-border-gray mb-4'>
      <div className='flex items-center p-2'>
        <Avatar image={userImage} size='small' highlight={false} />
        <span className='text-sm ml-1'>{username}</span>
      </div>
      <Image
        className='w-full object-cover aspect-square'
        src={image}
        alt={`photo by ${username}`}
        width={500}
        height={500}
        priority={priority}
        onClick={() => setOpenModal(true)}
      />
      <ActionBar
        username={username}
        text={text}
        createdAt={createdAt}
        likes={likes}
      />
      <CommentForm />
      {openModal && (
        <ModalPortal>
          <PostModal onClose={() => setOpenModal(false)}>
            <PostDetail post={post} />
          </PostModal>
        </ModalPortal>
      )}
    </article>
  );
}
