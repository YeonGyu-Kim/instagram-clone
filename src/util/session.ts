import { authOptions } from '@/app/api/auth/[...nextauth]/route';
import { getServerSession } from 'next-auth';
import { getToken } from 'next-auth/jwt';
import { NextRequest } from 'next/server';

export async function withSessionUser(
  handler: (sub?: string) => Promise<Response>,
  req: NextRequest
): Promise<Response> {
  const session = await getServerSession(authOptions);
  const user = session?.user;
  const secret = process.env.NEXTAUTH_SECRET;
  const token = await getToken({ req, secret });

  if (!user) {
    return new Response('Authentication Error', { status: 401 });
  }

  return handler(token?.sub);
}
