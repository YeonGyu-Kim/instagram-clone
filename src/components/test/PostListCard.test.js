import PostListCard from '@/components/PostListCard';
import { parseDate } from '@/util/date';
import { render, screen } from '@testing-library/react';
import { withRouter } from 'next/router';
import { Route } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('PostListCard', () => {
  const post = {
    username: 'lee',
    userImage:
      'https://lh3.googleusercontent.com/a/AGNmyxZN6KVGXD7LhM-Ppy6kKWHYJruXojxX_dngrV2Z=s96-c',
    image:
      '/_next/image?url=https%3A%2F%2Fcdn.sanity.io%2Fimages%2Fahsuynfv%2Fproduction%2F06dd22c42c5f030493eb025fd8b9f4d28fb807c4-1920x1080.jpg%3Fw%3D800&w=1080&q=75',
    likes: [],
    text: '풍경',
    comments: 8,
    id: '449fe9e7-b176-4a5c-80b5-01bcf501fa2a',
    createdAt: '2023-06-26T17:05:54Z',
  };
  it('포스트 카드 아이템 렌더링', () => {
    render(<PostListCard post={post} priority={false} />);

    //const image = screen.getAllByRole('img');
    expect(screen.getAllByText(post.username));
    expect(screen.getByText(post.text)).toBeInTheDocument();
    expect(screen.getByText(parseDate(post.createdAt))).toBeInTheDocument();
  });

  it('스냅샷 테스트', () => {
    const component = renderer.create(
      <PostListCard post={post} priority={false} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  // 배포 test
});
