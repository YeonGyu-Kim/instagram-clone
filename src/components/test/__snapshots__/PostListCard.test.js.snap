// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListCard 스냅샷 1`] = `
<article
  className="rounded-lg border border-border-gray mb-4"
>
  <div
    className="flex items-center p-4"
  >
    <div
      className="flex justify-center items-center rounded-full  w-9 h-9"
    >
      <img
        alt="user-profile"
        className="object-cover rounded-full false  w-[32px] h-[32px] p-[0.1rem]"
        referrerPolicy="no-referrer"
        src="https://lh3.googleusercontent.com/a/AGNmyxZN6KVGXD7LhM-Ppy6kKWHYJruXojxX_dngrV2Z=s96-c"
      />
    </div>
    <span
      className="text-sm ml-1"
    >
      lee
    </span>
  </div>
  <img
    alt="photo by lee"
    className="w-full object-cover aspect-square"
    data-nimg="1"
    decoding="async"
    height={500}
    loading="lazy"
    onClick={[Function]}
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fahsuynfv%252Fproduction%252F06dd22c42c5f030493eb025fd8b9f4d28fb807c4-1920x1080.jpg%253Fw%253D800%26w%3D1080%26q%3D75&w=1080&q=75"
    srcSet="/_next/image?url=%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fahsuynfv%252Fproduction%252F06dd22c42c5f030493eb025fd8b9f4d28fb807c4-1920x1080.jpg%253Fw%253D800%26w%3D1080%26q%3D75&w=640&q=75 1x, /_next/image?url=%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fahsuynfv%252Fproduction%252F06dd22c42c5f030493eb025fd8b9f4d28fb807c4-1920x1080.jpg%253Fw%253D800%26w%3D1080%26q%3D75&w=1080&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={500}
  />
  <div
    className="p-3 text-sm"
  >
    <div
      className="flex items-center justify-between"
    >
      <div
        className="flex gap-3"
      >
        <button
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={24}
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
            />
          </svg>
        </button>
        <svg
          fill="currentColor"
          height={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 32C114.6 32 0 125.1 0 240c0 47.6 19.9 91.2 52.9 126.3C38 405.7 7 439.1 6.5 439.5c-6.6 7-8.4 17.2-4.6 26S14.4 480 24 480c61.5 0 110-25.7 139.1-46.3C192 442.8 223.2 448 256 448c141.4 0 256-93.1 256-208S397.4 32 256 32zm0 368c-26.7 0-53.1-4.1-78.4-12.1l-22.7-7.2-19.5 13.8c-14.3 10.1-33.9 21.4-57.5 29 7.3-12.1 14.4-25.7 19.9-40.2l10.6-28.1-20.6-21.8C69.7 314.1 48 282.2 48 240c0-88.2 93.3-160 208-160s208 71.8 208 160-93.3 160-208 160z"
          />
        </svg>
      </div>
      <button
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 384 512"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M336 0H48C21.49 0 0 21.49 0 48v464l192-112 192 112V48c0-26.51-21.49-48-48-48zm0 428.43l-144-84-144 84V54a6 6 0 0 1 6-6h276c3.314 0 6 2.683 6 5.996V428.43z"
          />
        </svg>
      </button>
    </div>
    <div
      className="mt-1 mb-2"
    >
      <p>
        좋아요 0개
      </p>
      <p
        className="py-1"
      >
        <span>
          lee
        </span>
        <span
          className="ml-1 font-light"
        >
          풍경
        </span>
      </p>
      <button
        onClick={[Function]}
      >
        댓글 8개 모두 보기
      </button>
      <p>
        2 months ago
      </p>
    </div>
    <form
      className="flex items-center border-t border-border-gray p-2 text-sm"
      onSubmit={[Function]}
    >
      <svg
        fill="currentColor"
        height={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M4.111 2.18a7 7 0 1 1 7.778 11.64A7 7 0 0 1 4.11 2.18zm.556 10.809a6 6 0 1 0 6.666-9.978 6 6 0 0 0-6.666 9.978zM6.5 7a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm5 0a1 1 0 1 1-2 0 1 1 0 0 1 2 0zM8 11a3 3 0 0 1-2.65-1.58l-.87.48a4 4 0 0 0 7.12-.16l-.9-.43A3 3 0 0 1 8 11z"
          fillRule="evenodd"
        />
      </svg>
      <textarea
        autoComplete="off"
        autoCorrect="off"
        className="w-[90%] ml-2 bg-bg-gray focus:outline-none p-1 resize-none overflow-hidden"
        onChange={[Function]}
        placeholder="댓글 달기..."
        required={true}
        rows={1}
        value=""
      />
      <button
        className="text-btn-blue font-semibold w-[40px]"
        disabled={true}
      />
    </form>
  </div>
</article>
`;
